chapter2 update (#232)
liatrio/devops-bootcamp: 2-virtual-machines-containers/2.1-golden-images.md

![](img2/linux_light.svg ':size=100x100 :class=light-mode-icon')
![](img2/linux_dark.svg ':size=100x100 :class=dark-mode-icon')

?> If you have an M1 Mac checkout VirtualBox beta build for arm64. ( Note that as of 23/02/2023 the VirtualBox beta build for arm64 did not work)

?> assuming you have an m1 mac & the virtualbox beta is still not operational, the only m1 compatible section is the compatibility section.

### M1 Compatibility Excersize

1. You will need to install the [UTM](https://mac.getutm.app) project.

?> Assuming you have brew  `brew install --cask utm` should work

> 2. You will also need an ARM64 linux iso; ideally we want to use the [CentOS Stream 9](https://www.centos.org/download/) iso located  make sure to select the *ARM64* version.

Reviewer: This reads a little strangely. Maybe something like

```suggestion
2. You will also need an ARM64 linux iso; ideally we want to use the [CentOS Stream 9](https://www.centos.org/download/) iso.  Make sure to select the *ARM64* version.
```

> 2. You will also need an ARM64 linux iso; ideally we want to use the [CentOS Stream 9](https://www.centos.org/download/) iso located  make sure to select the *ARM64* version.

Author: https://github.com/liatrio/devops-bootcamp/pull/232/commits/1a1d5614c2cac432c1f7c49256ab9d684c371880

> 3. Using UTM, create a new virtual machine and install CentOS using the ISO you downloaded.

Reviewer: Inconsistent casing with `iso`. I I have seen it capitalized in other parts of the boot camp like [here](https://devops-bootcamp.liatr.io/#/2-virtual-machines-containers/2.1-golden-images?id=exercise-1-manual-build). But above in step 2 above you have it lower case. I don't really care either way I would just like us to be consistent.

Author: https://github.com/liatrio/devops-bootcamp/pull/232/commits/5e2b6d218128c803025422540e06775295e52bc6

![](img2/linux_light.svg ':size=100x100 :class=light-mode-icon')
![](img2/linux_dark.svg ':size=100x100 :class=dark-mode-icon')

> ?> If you have an m1 mac checkout virtualbox beta build for arm64. ( Note that as of 23/02/2023 the virtualbox beta build for arm64 did not work)

Reviewer: ```suggestion
?> If you have an M1 Mac checkout VirtualBox beta build for arm64. ( Note that as of 23/02/2023 the VirtualBox beta build for arm64 did not work)
```

Author: fixed here: https://github.com/liatrio/devops-bootcamp/pull/232/commits/4ed1bd2e10647701469571f6e8c26695842ca0ff

?> assuming you have an m1 mac & the virtualbox beta is still not operational, the only m1 compatible section is the compatibility section.

> ### M1 Compatibility Excersize

Reviewer: Is there a reason you kept the VirtualBox section? I have not gone through the whole Bootcamp yet but I feel like we never go back to virtual box and we might just be able to change UTM.

Reasons to keep both that I can think of:
1) VirtualBox is cross platform UTM is not

Reasons to get rid of VirtualBox:
1) In the Introduction we already specify this Bootcamp is written for Mac users so the reason to keep it might not matter
2) Now we need to maintain two sets of instructions.

I searched through the bootcamp and only saw virtualbox being referenced in chapter 2. With that I feel like we should just switch to UTM. I am assuming that UTM works well across intel and apple silicon macs. And I see this section really as a way to just introduce the concept of virtual machines and get some hands on with it.

Reviewer: It feels like you went into too much detail in this section. The previous instructions were
>Using [VirtualBox](https://www.virtualbox.org/wiki/Downloads), create a new virtual machine and install CentOS using the ISO you downloaded.

I am inclined to say this should just say something like
>Using [UTM](https://mac.getutm.app/), create a new virtual machine and install CentOS using the ISO you downloaded.

I think this holds their hand less, will be easier to maintain down the road (UIs change, and specific instructions are prone to going stale). Additionally this sends the user to a page about UTM where they can read about it. Personally this was software I had never heard of so when asked to just `brew` install UTM I was like "what is it and why am I installing it"

Reviewer: After discussing this further with Evan I am convinced we should keep both the VirtualBox and UTM sections in. UTM might get cleaned up down the road when VirtualBox has a working build for arm processors. And I see this section as a way to introduce Apprentices to virtualization and we move past it pretty quickly. 

I do think we could simplify the step-by-step instructions to be more inline with the existing VirtualBox instructions but this is not a blocker.

Author: this should be fixed in new commit

> 1. You will need to install the utm project. assuming you have brew  `brew install --cask utm` should work

Reviewer: ```suggestion
1. You will need to install the UTM project. assuming you have brew  `brew install --cask utm` should work
```

Reviewer: Can we add a link to the UTM project here also for curious apprentices who want to know _what_ they are installing

> 2. You will also need an ARM64 linux iso; ideally we want to use the centos9 iso located [here](https://mirrors.centos.org/mirrorlist?path=/9-stream/BaseOS/aarch64/iso/CentOS-Stream-9-latest-aarch64-dvd1.iso&redirect=1&protocol=https) make sure to select the ARM64 version.

Reviewer: The directions make it seem like the user will have to find and select the right iso, but we directly link an iso. Mirror links like this are often break. I think we should just link back to https://www.centos.org/download/ as that will likely not be changing and if it does I imagine they will handle redirecting for a while.

Reviewer: Additionally you link to the full blown centOS download. I think we could probably get away with the minimal image

Reviewer: I see why you just linked the iso now. This changes from centOS Linux to CentOS Stream. And it appears that on the Stream releases there is only a single version. Kind of crummy to have to download a 7GB file for this exercise. That fact alone is making me think we just target CentOS Linux 7 but with the EOL next year I could see this going either way.

Reviewer: Well in testing I ran into issues booting a CentOS 7 minimal iso in UTM so Stream is fine with me

Reviewer: I do think it is worth adding an info section to advise users to change the software option in the GUI install to be 'server'. Correct me if I am wrong but I think the default options will give you a GUI experience of CentOS and I imagine we are more likely to be dealing with GUI-less versions with clients (or I hope so)

Author: this should be fixed in https://github.com/liatrio/devops-bootcamp/pull/232/commits/a7d01112b7035bc6782caf367e6a7c486cde8787

3. Once you have the iso downloaded open the UTM gui.
    1. create a new vm, select virtualize
    2. when prompted for "virtualization engine", you need to give your vm the iso you downloaded. probably in the local Downloads directory.
    ![](img2/UTM_iso_selection.png)
    3. Once you have an iso selected, continue to the end (default vm hard settings will work fine).
    4. Launch your vm, go through the centos9 install. (defaults/automatic should be fine for everything)
    5. Once you asks you to reboot shut the vm down and unmount your iso before rebooting.
    ![](img2/UTM_iso_unmount.png)

> 4. Congradulations you have a centos9 image, you can create ssh keys manually with the same steps as below

Reviewer: ```suggestion
4. Congratulations you have a CentOS9 VM. You can create ssh keys manually with the same steps as below.
```

Author: fixed here: https://github.com/liatrio/devops-bootcamp/pull/232/commits/4ed1bd2e10647701469571f6e8c26695842ca0ff

#### Exercise 1: Manual Build

For this exercise, you will create a Golden Image by manually installing and configuring CentOS 7 in a VirtualBox VM.

> 1. Download the [CentOS 9](https://mirrors.centos.org/mirrorlist?path=/9-stream/BaseOS/aarch64/iso/CentOS-Stream-9-latest-aarch64-dvd1.iso&redirect=1&protocol=https) ISO for a 64-bit x86 machine.

Reviewer: Same concern about a specific link to a mirror download.

Author: fixed here: https://github.com/liatrio/devops-bootcamp/pull/232/commits/4216c2744c7c19579530aaeddbf63b1d30d7636c
